libraries/standard/mqtt/src/mqtt.c

```diff
@@ -365,15 +402,21 @@ static MQTTStatus_t discardPacket( const MQTTContext_t * pContext,
     MQTTStatus_t status = MQTTRecvFailed;
     int32_t bytesReceived = 0;
     size_t bytesToReceive = 0U;
-    uint32_t totalBytesReceived = 0U, entryTimeMs, elapsedTimeMs;
+    uint32_t totalBytesReceived = 0U, entryTimeMs = 0U, elapsedTimeMs = 0U;
     uint32_t remainingTimeMs = timeoutMs;
     MQTTGetCurrentTimeFunc_t getTimeStampMs = NULL;
     bool receiveError = false;
 
     assert( pContext != NULL );
     bytesToReceive = pContext->networkBuffer.size;
     getTimeStampMs = pContext->callbacks.getTime;
-    entryTimeMs = getTimeStampMs();
+
+    assert( ( getTimeStampMs != NULL ) || ( timeoutMs == 0U ) );
+
+    if( getTimeStampMs != NULL )
```

Comment:
See previous comment about verifying `pContext->callbacks.getTime` during construction rather than before every time it is used.